### DISCUSSION TITLE
appwrite with react query

### DISCUSSION THREAD

**Discussion by stephane2018** [appwrite/appwrite#4552] [General]
is it possible to use appwrite with react query library?

**Comment by stnguyen90** [comment]
Yes, of course! With React Query, you specify a [Query Function](https://tanstack.com/query/v4/docs/guides/query-functions) that will do the actual fetching. This function would be where you put your code that uses the Appwrite SDK to make the API call.

**Reply by stephane2018** [reply]
ok now concerning the caching process of data it is not the redoudant ? since appwrite sdk have already caching system embeded, and react query too.

**Reply by stnguyen90** [reply]
Appwrite doesn't have any caching in the SDK of the data. It does have persistence of the session, though.

**Reply by stephane2018** [reply]
ok
